src/ppx/grfx/grfx_command.cpp

```diff
@@ -257,5 +257,239 @@ void CommandBuffer::Draw(const grfx::FullscreenQuad* pQuad, uint32_t setCount, c
+    uint32_t                       bufferOffset,
+    const grfx::Buffer*            pBuffer)
+{
+    PushDescriptorImpl(
+        grfx::COMMAND_TYPE_GRAPHICS,          // pipelineBindPoint
+        pInterface,                           // pInterface
+        grfx::DESCRIPTOR_TYPE_UNIFORM_BUFFER, // descriptorType
+        binding,                              // binding
+        set,                                  // set
+        bufferOffset,                         // bufferOffset
+        pBuffer,                              // pBuffer
+        nullptr,                              // pSampledImageView
+        nullptr,                              // pStorageImageView
+        nullptr);                             // pSampler
+}
+
+void CommandBuffer::PushGraphicsStructuredBuffer(
+    const grfx::PipelineInterface* pInterface,
+    uint32_t                       binding,
+    uint32_t                       set,
+    uint32_t                       bufferOffset,
+    const grfx::Buffer*            pBuffer)
+{
+    PushDescriptorImpl(
+        grfx::COMMAND_TYPE_GRAPHICS,                // pipelineBindPoint
+        pInterface,                                 // pInterface
+        grfx::DESCRIPTOR_TYPE_RO_STRUCTURED_BUFFER, // descriptorType
+        binding,                                    // binding
+        set,                                        // set
+        bufferOffset,                               // bufferOffset
+        pBuffer,                                    // pBuffer
+        nullptr,                                    // pSampledImageView
+        nullptr,                                    // pStorageImageView
+        nullptr);                                   // pSampler
+}
+
+void CommandBuffer::PushGraphicsStorageBuffer(
+    const grfx::PipelineInterface* pInterface,
+    uint32_t                       binding,
+    uint32_t                       set,
+    uint32_t                       bufferOffset,
+    const grfx::Buffer*            pBuffer)
+{
+    PushDescriptorImpl(
+        grfx::COMMAND_TYPE_GRAPHICS,                // pipelineBindPoint
+        pInterface,                                 // pInterface
+        grfx::DESCRIPTOR_TYPE_RW_STRUCTURED_BUFFER, // descriptorType
+        binding,                                    // binding
+        set,                                        // set
+        bufferOffset,                               // bufferOffset
+        pBuffer,                                    // pBuffer
+        nullptr,                                    // pSampledImageView
+        nullptr,                                    // pStorageImageView
+        nullptr);                                   // pSampler
+}
+
+void CommandBuffer::PushGraphicsSampledImage(
+    const grfx::PipelineInterface* pInterface,
+    uint32_t                       binding,
+    uint32_t                       set,
+    const grfx::SampledImageView*  pView)
+{
+    PushDescriptorImpl(
```

Comment:
Like, here.  In D3D12, have PushDescriptorImpl() expand to a `static_assert` or some such.

Reply:
See previous comment.